tests/test_unit/test_auth/test_azure_token_provider.py

```diff
@@ -0,0 +1,8 @@
+from odp_sdk.auth import AzureTokenProvider
+
+
+def test_get_token(azure_token_provider: AzureTokenProvider):
+    access_token = azure_token_provider.authenticate()
+
+    assert access_token
+    assert access_token["access_token"]
```

Comment:
Maybe assert that the correct http-request was made?

Reply:
Since I didn't set `assert_all_requests_are_fired` field while creating the mock, it automatically fails if the request I mocked is not called.

Reply:
That's fine, but you can still assert that a single request is sent in order ensure that the access token was in fact retrieved.

Inversely for getting the token before the previous one expired - you can assert that a request wasn't sent

Reply:
I added the check